src/catkin_pkg/packages.py

```diff
@@ -129,25 +129,27 @@ def find_packages_allowing_duplicates(basepath, exclude_paths=None, exclude_subs
     if not data:
         return {}
 
-    if len(data) < 100:
+    try:
+        if len(data) < 100:
+            raise Error
+        parser = _PackageParser(warnings is not None)
+        pool = multiprocessing.Pool()
+        try:
+            path_parsed_packages, warnings_lists = zip(*pool.map(parser, data))
+        finally:
+            pool.close()
+            pool.join()
+        if parser.capture_warnings:
+            map(warnings.extend, warnings_lists)
+        return dict(path_parsed_packages)
+    except:
```

Comment:
Roughly yes. But you don't need to move part of the logic into a function in order to do that. I would try to do a conditional logic like this:

```
parallel = len(data) > 100
if parallel:
    try:
        pool = multiprocessing.Pool()
    except OSError:  # is this the only possible exception?
        parallel = False

if parallel:
    # use multiprocessing pool
else:
    # use sequential loop
```